traitschema/schema.py

```diff
@@ -222,8 +231,39 @@ def from_hdf(cls, filename):
         self = cls()
         with h5py.File(filename, 'r') as hfile:
             for name in self.visible_traits():
+                trait = self.trait(name)
+                if name not in hfile:
+                    continue
                 data = hfile['/{}'.format(name)].value
+
+                # ndarrays with a single type do not return anything for
+                # .dtype.names, so this is a reasonable way to check if the
+                # ndarray should really be a recarray that has multiple types
```

Comment:
Why not store an attribute in the HDF5 array that indicates that it's a recarray?